cocos/core/platform/debug.ts

```diff
@@ -49,80 +50,68 @@ let ccAssert = (condition: any, message?: any, ...optionalParams: any[]): void =
 let ccDebug = ccLog;
 
 /**
- * @en Format a string.
- * @zh 格式化字符串。
- * @param message @zh 包含零个或多个需要替换的JavaScript字符串。@en JavaScript objects to replace substitution strings in msg.
- * @param optionalParams  @zh 用来替换在message中需要替换的JavaScript对象。@en JavaScript objects with which to replace substitution strings within msg.
+ * Constructs a string from a sequence of js object arguments.
  */
-function formatString (message?: any, ...optionalParams: any[]): any {
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-    return legacyCC.js.formatStr.apply(null, [message].concat(optionalParams));
+function formatString (...data: unknown[]): string {
+    return legacyCC.js.formatStr.apply(null, data) as string;
 }
 
 /**
- * @en Outputs a message to the Cocos Creator Console (editor) or Web Console (runtime). This gives you additional control over the format of the output.
- * @zh 输出一条消息到 Cocos Creator 编辑器的 Console 或运行时 Web 端的 Console 中。这为你提供了对输出格式的额外控制。
- * @param message @zh 包含零个或多个需要替换的JavaScript字符串。@en JavaScript objects to replace substitution strings in msg.
- * @param optionalParams  @zh 用来替换在message中需要替换的JavaScript对象。@en JavaScript objects with which to replace substitution strings within msg.
+ * @en Outputs a log message to the console. The message may be a single string (with optional substitution values), or it may be any one or more JavaScript objects.
+ * @zh 向控制台输出一条日志信息。这条信息可能是单个字符串（包括可选的替代字符串），也可能是一个或多个对象。
  */
-export function log (message?: any, ...optionalParams: any[]): void {
-    return ccLog(message, ...optionalParams);
+export function log (...data: unknown[]): void {
+    return ccLog(...data);
 }
```

Comment:
Our implementation is aligned to web standards, so the log method should be defined as `function log (...data: unknown[]): void`, the same goes for other methods like `warn error debug`